js/src/forum/components/IndexPage.js

```diff
@@ -33,26 +33,25 @@ export default class IndexPage extends Page {
     // probably want to refresh the results. We will clear the discussion list
     // cache so that results are reloaded.
     if (app.previous instanceof IndexPage) {
-      app.cache.discussionList = null;
+      app.cache.discussionList.clear();
     }
 
     const params = this.params();
 
-    if (app.cache.discussionList) {
+    if (app.cache.discussionList.hasDiscussions()) {
       // Compare the requested parameters (sort, search query) to the ones that
       // are currently present in the cached discussion list. If they differ, we
-      // will clear the cache and set up a new discussion list component with
-      // the new parameters.
+      // will clear the cache update the parameters.
       Object.keys(params).some((key) => {
-        if (app.cache.discussionList.props.params[key] !== params[key]) {
-          app.cache.discussionList = null;
+        if (app.cache.discussionList.getParams()[key] !== params[key]) {
+          app.cache.discussionList.clear();
```

Comment:
In general, we'd like to redraw when clearing, but in this case, it's true that we get an unnecessary redraw. Since `refresh` has `clear` as a parameter, instead of clearing if this condition is met, we can set a constant which we will pass into that refresh call.